File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int size = nums.length;
        int[] leftArray = new int[size];
        int[] rightArray = new int[size];
        
        leftArray[0] = 1;
        for (int i = 1; i < size; i++) {
            leftArray[i] = leftArray[i - 1] * nums[i - 1];
        }
        
        rightArray[size - 1] = 1;
        for (int i = size - 2; i >= 0; i--) {
            rightArray[i] = rightArray[i + 1] * nums[i + 1];
        }
        
        int[] answer = new int[size];
        for (int i = 0; i < size; i++) {
            answer[i] = leftArray[i] * rightArray[i];
        }
        
        return answer;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const N = nums.length
    
    const leftArray = Array(N).fill(1)
    const rightArray = Array(N).fill(1)
    
    for (let i = 1; i < N; i++) {
        leftArray[i] = leftArray[i - 1] * nums[i - 1]
    }
    
    for (let i = N - 2; i >= 0; i--) {
        rightArray[i] = rightArray[i + 1] * nums[i + 1]
    }
    
    const answer = []
    
    for (let i = 0; i < N; i++) {
        answer[i] = leftArray[i] * rightArray[i]
    }
    
    return answer
};